db-utils/src/main/kotlin/com/icerockdev/common/database/sql/ColumnType.kt

```diff
@@ -40,8 +38,27 @@ class GeographyPointColumnType(private val length: Int = 4326) : ColumnType() {
             else -> error("$value of ${value::class.qualifiedName} is not valid geo object")
         }
     }
+}
 
+@Suppress("UNCHECKED_CAST")
+open class PGEnumColumnType<T : Enum<T>>(
+    private val enumName: String,
+    private val clazz: Class<T>,
+    private val enumType: String? = null
+) : ColumnType() {
+    override fun sqlType(): String = enumType ?: "${enumName}_enum"
+    override fun valueFromDB(value: Any): Any {
+        return java.lang.Enum.valueOf(clazz, value as String)
```

Comment:
here we can use with kotlin this - `clazz.java.enumConstants.first { it.name == value }`
just like in `org.jetbrains.exposed.sql.EnumerationNameColumnType`